commonDependencyVersionConstraints/build.gradle

```diff
@@ -129,6 +129,7 @@ dependencies {
     implementation group: 'org.jsoup', name: 'jsoup', version: '1.15.3'
     implementation group: 'org.apache.tika', name: 'tika-core', version: '1.28.4'
     implementation group: 'com.jayway.jsonpath', name: 'json-path', version: '2.9.0'
+    api group: 'org.apache.httpcomponents', 'name': 'httpclient', version: '4.5.13'
```

Comment:
Yeah, I started with `implementation`, but that didn't actually fix the issue--the gradle dependency analyzer showed that 4.5.12 was still being pulled in in several places. Changing it to `api` fixed all cases. My theory was that because the parent dependency was `api`, that's why switching this to `api` worked to fix it. I don't know if that's a correct mental model of how gradle works or not.

Reply:
Hmm, I'll look into this today.  I'm concerned that there is some dependency that isn't correctly mapped to this set of build constraints

Reply:
My understanding is that we want to use API on all the constraints in this file. This is saying, require at least this version in the compile and runtime path, and if not, recompile with at least this version resolved.

This is different from including API vs Implementation in dependencies in a sub-project which affect how consumers of that library treat that dependency

Reply:
This is ~bug~ _quirk_ of how test fixtures work.  Because of how they are grafted onto the java build system they don't always operate the same as normal java application and test dependencies 😱 

- https://github.com/gradle/gradle/issues/13067